### DISCUSSION TITLE
Consider progressive enhancement

### DISCUSSION THREAD

**Discussion by selfhoster1312** [torrust/torrust-index-gui#441] [Feature requests]
I've had trouble running the frontend so far, and it's not because of the backend. It's really hard to make a "good" JS UI and still it can break accessibility and navigation, and JS can be broken on the client side for many reasons and just not display anything.

It would be great to consider to provide all the functionality without javascript, then sprinkle some javascript on top for animations and whatnot (or no JS at all), what we call "progressive enhancement" in the web world. I'm not sure how/if that's possible with Nuxt/Vue but it's certainly possible in general, prevents many bugs and makes everyone happy to have pages loading much faster than waiting for megabytes of JS to download and execute.

**Comment by josecelano** [comment]
Hi @selfhoster1312 thank you for your feedback. It seems there has been some work in NUXT, at least for forms: https://github.com/nuxt/nuxt/pull/20852

I agree that making a good UI is a challenging task. It would be nice to have such a simple version, but I think if you manage to build a "good" JS UI without breaking accessibility and navigation, the UX would be definitively better using JS. But I'm not a UX designer or even a front-end developer. Maybe we could have an alternative version, but that would be too costly to maintain. Apart from forms, I will look at other possibilities to apply "progressive enhancements" to the current solution. Let me know if you know more concrete examples, demo repos, etcetera, following that approach. We are focused on releasing a major version, so we are mainly writing documentation, fixing bugs, and testing, ...

Regarding the app size, it's still small:

![image](https://github.com/torrust/torrust-index-frontend/assets/58816/c6de6f3c-12f0-4e79-a108-476ab6168b5e)
